### PR TITLE
Cord v2

### PR DESCRIPTION
**Pull request by hrhino**
Continues #1633 at request of @edmundnoble. `Cord` work is all his.

### CODE REVIEW

**Comment by jdegoes** [comment]
@hrhino Is this good to merge?

**Comment by hrhino** [comment]
I think so, but that's why I put it up 😉. The `Cord` implementation is verbatim from #1633.

**Comment by hrhino** [comment]
So it didn't actually work. Fixed now.

**Comment by hrhino** [comment]
Now it is, IMHO, g2g.

File: base/shared/src/main/scala/scalaz/debug/debug.scala [hunk 1]
```diff
@@ -3,66 +3,97 @@ package debug
...
   }
 
   implicit final def debugInterpolator(sc: StringContext): DebugInterpolator.Interpolator =
     new DebugInterpolator.Interpolator(sc)
 }
 
-object DebugInterpolator {
-  final class HasDebug private (override val toString: String) extends AnyVal
+object DebugInterpolator extends DebugInterpolator {
+  implicit def mkHasDebug[A](x: A)(implicit D: Debug[A]): HasDebug =
```
Comment (alexknvl) [inline base/shared/src/main/scala/scalaz/debug/debug.scala]: This is a cool trick. I would normally do `Instance[Debug]`, which unnecessarily delays `shows`.

Comment (hrhino) [reply base/shared/src/main/scala/scalaz/debug/debug.scala]: This is @edmundnoble's suggestion. And yes, it is.

**Comment by hrhino** [comment]
If we want to merge this soon I'll make a follow-up ticket, otherwise just hold tight

**Comment by hrhino** [comment]
@edmundnoble get off your ass and give us ztest

**Comment by hrhino** [comment]
Thanks Ed.

@jdegoes Is this good to merge?

**Pull request merged**